temporalio/worker/_workflow.py

```diff
@@ -250,9 +316,32 @@ async def _handle_activation(
                         activate_task, self._deadlock_timeout_seconds
                     )
                 except asyncio.TimeoutError:
-                    raise RuntimeError(
-                        f"[TMPRL1101] Potential deadlock detected, workflow didn't yield within {self._deadlock_timeout_seconds} second(s)"
-                    )
+                    # extract the stack of the workflow if possible
```

Comment:
So this `_handle_activation` method is getting large and deadlock is a rare piece of it to take such a large percentage of the mental model. Would love:

```python
raise RuntimeError(
    f"[TMPRL1101] Potential deadlock detected, workflow didn't yield within {self._deadlock_timeout_seconds} second(s)"
)
```

to just change to:

```python
raise self.build_deadlock_exception(workflow)
```

or something